mlflow/gateway/providers/anthropic.py

```diff
@@ -220,11 +223,29 @@ def __init__(self, config: RouteConfig) -> None:
         if config.model.config is None or not isinstance(config.model.config, AnthropicConfig):
             raise TypeError(f"Invalid config type {config.model.config}")
         self.anthropic_config: AnthropicConfig = config.model.config
-        self.headers = {
+
+    @property
+    def headers(self) -> dict[str, str]:
+        return {
             "x-api-key": self.anthropic_config.anthropic_api_key,
             "anthropic-version": self.anthropic_config.anthropic_version,
         }
-        self.base_url = "https://api.anthropic.com/v1/"
+
+    @property
+    def base_url(self) -> str:
+        return "https://api.anthropic.com/v1/"
+
+    @property
+    def adapter(self):
+        return AnthropicAdapter
```

Comment:
Do we need this? Seems AnthropicProvider extends from AnthropicAdapter, so we can directly call `chat_to_model` method?

Reply:
It is pretty strange but only `AnthropicProvider` extends its adapter, none of other providers does it😅